tools/cloud-build/images/cluster-toolkit-dockerfile/README.md

```diff
@@ -0,0 +1,84 @@
+# Cluster Toolkit Dockerfile
+
+This repository contains a Dockerfile for building a Docker image with Cluster Toolkit and its dependencies installed and the `gcluster` binary readily available for use.
+
+## System Requirements
+
+* [Docker Engine](https://docs.docker.com/engine/) needs to be installed.
+
+## Environment Variables
+The following environment variables can be used to customize the build process:
+* **`BASE_IMAGE`**: The base image to use for the build. Defaults to `gcr.io/google.com/cloudsdktool/google-cloud-cli:stable`.
+* **`TERRAFORM_VERSION`**: The version of Terraform to install. Defaults to `1.5.2`.
+* **`PACKER_VERSION`**: The version of Packer to install. Defaults to `1.8.6`.
+* **`GO_VERSION`**: The version of Go to install. Defaults to `1.21.0`.
+* **`CLUSTER_TOOLKIT_VERSION`**: The version (tag) of the Cluster Toolkit to install. Defaults to the current version associated with the `main` branch of the cluster toolkit repository.
+
+## Build the Docker Image
+To build the Docker image, navigate to the directory the Dockerfile is present in and run the following command:
+
+```bash
+docker build --build-arg BASE_IMAGE=<base_image> \
+             --build-arg TERRAFORM_VERSION=<terraform_version> \
+             --build-arg PACKER_VERSION=<packer_version> \
+             --build-arg GO_VERSION=<go_version> \
+             --build-arg CLUSTER_TOOLKIT_VERSION=<cluster_toolkit_version> \
+             -t <image_name> .
+```
+
+Example:
+
+```bash
+docker build --build-arg CLUSTER_TOOLKIT_VERSION=v1.40.0 -t my-cluster-toolkit-image .
+```
+
+The above example builds an image tagged `my-cluster-toolkit-image` and sets the CLUSTER_TOOLKIT_VERSION to v1.40.0 while using the default values for other arguments.
+
+## Run the Docker Image
+To run the Docker image, use the following command:
+
+```bash
+docker run <image_name> <gcluster_command>
```

Comment:
Let's describe the basic usage in a top-down way instead of bottom-up.  By this, I mean:
1. Show the `docker run` command which mounts both the gcloud directory and the user's current directory (as a data directory).
2. Break down each part of the command to explain why it's important.

I think this approach would be better because both mounts are pretty important for even basic usage.